handlers/life_journey.go

```diff
@@ -28,22 +30,97 @@ func (h *Handler) LifeJourney(c echo.Context) error {
 		c.Logger().Error(err)
 		return c.JSON(http.StatusBadRequest, lifeJourneyResponse)
 	}
-	ljList, err := lifejourneys.LifeJourneyService.LoadLifeJourneys()
+
+	var lifeJourneyList []models.LifeJourney
+	lifeJourneyList, err := getLifeJourneyBenefits(lifeJourneyRequest.Id)
 	if err != nil {
 		c.Logger().Error(err)
 		return c.JSON(http.StatusBadRequest, lifeJourneyResponse)
 	}
 
-	if lifeJourneyRequest.Id != "" {
+	lifeJourneyResponse.LifeJourneyList = lifeJourneyList
+	return c.JSON(http.StatusOK, lifeJourneyResponse)
+}
+
+// LifeJourneyBenefits
+// @Summary Request Info on Life Journey Related Benefits
+// @ID life-journey-benefits
+// @Accept  json
+// @Produce json
+// @Success 200 {object} []models.Benefits
+// @Failure 400 {object} renderings.BenefitServiceError
+// @Failure 404 {object} renderings.BenefitServiceError
+// @Failure 500 {object} renderings.BenefitServiceError
+// @Router /lifejourneys/:id/benefits [get]
+func (h *Handler) LifeJourneyBenefits(c echo.Context) error {
+	var lifeJourneyBenefitsResponse []models.Benefits
+	lifeJourneyBenefitsRequest := new(bindings.LifeJourneyBenefitsRequest)
+
+	// Bind the request into our request struct
+	if err := c.Bind(lifeJourneyBenefitsRequest); err != nil {
+		c.Logger().Error(err)
+		return c.JSON(http.StatusBadRequest, lifeJourneyBenefitsResponse)
+	}
+
+	if lifeJourneyBenefitsRequest.Id == "" {
+		return c.JSON(http.StatusUnprocessableEntity, lifeJourneyBenefitsResponse)
+	}
```

Comment:
This should probably return an http 400 rather than a 422 response. Also log the error.

Reply:
I added a log but there is no error that would be thrown. I added a string to the log to say what the error would be